src/storage/index.ts

```diff
@@ -0,0 +1,79 @@
+import express = require('express')
+import { server } from 'decentraland-server'
+
+import * as multer from 'multer'
+import * as multerS3 from 'multer-s3'
+
+import {
+  bucketName,
+  checkFile,
+  parseFileBody,
+  readFile,
+  s3,
+  uploadFile
+} from '../S3'
+
+const ENTRY_FILENAME = 'entry.json'
+const MAX_FILE_SIZE = 3000000
+
+export enum EntryPrefix {
+  Contest = 'contest',
+  Project = 'project'
+}
+
+interface ContentTypeExtension {
+  [key: string]: string
+}
+
+const contentTypeExtensionMap: ContentTypeExtension = {
+  'image/png': 'png',
+  'video/webm': 'webm'
+}
+
+export function getEntryKey(projectId: string, prefix: EntryPrefix): string {
+  return prefix + '/' + projectId + '/' + ENTRY_FILENAME
+}
+
+export async function readEntry(projectId: string, prefix: EntryPrefix) {
+  try {
+    const key = getEntryKey(projectId, prefix)
+    const file = await readFile(key)
+    return parseFileBody(file)
+  } catch (error) {
+    // No previous entity
+  }
+}
+
+export async function saveEntry(
+  projectId: string,
+  entry: any,
```

Comment:
can't this be of type Entry?

Reply:
This function serves both for ProjectEntry and ContestEntry they currently don't share a common parent type. I will make it inherit from a Base type so we can enforce the type validation.  👍